### ISSUE TITLE
Part-time Jobs

### ISSUE THREAD

**Issue by exectails** [aura-project/aura#237]
Split from quests, since this will take some more time, and it's important for people to see that we still need a lot of PTJs logged.

Reference: http://wiki.mabinogiworld.com/index.php?title=Part-time_Jobs&oldid=30265

**Part-time jobs**
- [x] **Tir**
- [x] Church
- [x] Grocery Store
- [x] Neighboring
- [x] Healer
- [x] Blacksmith
- [x] General Store
- [x] Inn
- [x] **Dugald Aisle**
- [x] Neighboring
- [x] **Dunbarton**
- [x] Book Store
- [x] Town
- [x] Glenis
- [x] Church
- [x] Healer
- [x] Weapon Shop
- [x] Tailor
- [x] Library
- [x] General Store
- [x] **Bangor**
- [x] Church
- [x] Blacksmith (Elen)
- [x] Mine

**Comment by DeleteIT** [comment]
This is what is required for this particular issue:
http://aura-project.org/forum/topic/173-ptj-dialogues/

**Comment by exectails** [comment]
For those who don't frequently visit the forum, we have a list over there, of what has to be logged for the PTJs to be implemented. Check it out if you'd like to help: http://aura-project.org/forum/topic/173-ptj-dialogues/

**Comment by gcscaglia** [comment]
Bellow are the full logs for Piaras PTJ. These are absolutely *RAW*. There are unrelated packets in the logs, including other PTJs I was doing in between the runs.

[Piaras.zip](https://github.com/aura-project/aura/files/191234/Piaras.zip)

The logs include all possible dialogs (out of order), which are:
* Dialog for when you're already doing a different PTJ.
* Dialog for coming back early without being done.
* Dialog for coming back early after being done.
* Report later dialog.
* Report now and cancel when choosing reward.
* Report now and choose reward.
* Dialog for reporting without having done anything.
* Dialog for reporting with having only done a little.
* Dialog for reporting with having done almost everything.
* Dialog for reporting with having done everything.
* Dialog for PTJ keyword while no PTJs are given out (Before Time).
* Dialog for PTJ keyword while no PTJs are given out (After Time).
* Dialog for declining the PTJ offer.

I'm writing a MabiPale2 plugin to automate logging PTJs and plan on write the actual PTJs afterwards, but I'm posting the logs here in case I end up MIA before that.

**Comment by Pegwald** [comment]
Training skills like Tailoring and Blacksmithing is a pain without so many missing PTJs, so I will just leave this very innocent (yet very obvious) comment here to remind people that I would very much like some more PTJs soon. ^^''

**Comment by exectails** [comment]
Did your plugin/script plans get anywhere, @gcscaglia?

**Comment by valdeza** [comment]
So how did I do on [getting Deian's dialogue](https://github.com/valdeza/aura/tree/resrch_ptj-dialogue/etc/ptj-dialogue/tir-chonaill/deian) ([filename key](https://github.com/valdeza/aura/blob/resrch_ptj-dialogue/etc/ptj-dialogue/expected-dataset.txt))?
For some reason, I can't upload a .zip file of all that on here, so I put it on my fork: [link](https://github.com/valdeza/aura/blob/resrch_ptj-dialogue/etc/ptj-dialogue/tir-chonaill/deian/deian.zip?raw=true)

This process really takes a while, and on the forums you seem to prefer having the packet data.
With MabiPale, do you think it's not a plugin that we need, but rather some core functionality, such as an include filter (versus the current exclude) or an option to export a selected time span?
At the moment, we're only grabbing NPC dialogue, but what if we find ourselves needing to search for something else in the future? While I'm thinking of forking MabiPale and trying to add that functionality, I can't exactly promise it actually being done (summer classes and all).

**Comment by exectails** [comment]
Personally I'm a little paranoid about whitelist-filtering while packet logging, because you don't know what you're missing. You might be thinking "I *only* need packet XY", but then something unforeseen happens, you only notice it after you analyze your logs, and suddenly you have to log everything again, because things happened that you didn't expect. This is especially dangerous if the user logging is a newbie. Though a search function might be good =|

The reason I prefer the raw logs is that we get the original data, without any risk of something being wrong. Additionally, Pale can work with that data. For example, for your log "inquire-ptj" the packet analyzer plugin generates the following code:
```csharp
Msg(L("It's not time for part-time jobs yet.<br/>Come back later."));
```
Or if someone released a dialog parser plugin tomorrow, or the dialog conventions in Aura change, we can use old logs and just run them through it. That wouldn't be possible if we had only the phrases. Conveniently, just uploading the logs is easier for the contributor as well.

Anyway, your logs look pretty good :+1:

**Comment by exectails** [comment]
A pull request for each individual feature/fix/etc would be good. I'm not sure what you mean about timestamps, the logs do include the time, as that's important to identify when a packet was sent, and as response to what, or simply out of the blue.

**Comment by valdeza** [comment]
Say we wanted to find what Simon said when you turned in your PTJ. Even with our new search function, we will have to search through nearly everything Simon has said if we happened to choose a bad spot to start our search.

But if our fellow recorder provided sufficiently detailed action timestamps during their recording, we would know that they turned in the PTJ at time 13:51:12.432-04:00, so we'll start our search there. It's a form of annotation, if you will.

**Comment by exectails** [comment]
Oh, you mean the Erinn time. I don't think we need timestamps for that, just calculate the Erinn time based on the time from the packet.

**Comment by valdeza** [comment]
Huh. I never thought of converting real time to Erinn time. 

Still, I didn't mean Erinn time! D:
I'm talking about something like this:

\> you've been doing PTJs for everyone in Dunbarton multiple times, so everyone's words are fairly blended
\> return to Simon
\> before you report now, you switch to the timestamping script and write a quick message
> \> Turned in 3 star job to Simon
>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;@ 13:51:12.432-04:00 <- (this is automatically generated after you hit Enter)

`13:51:12.432-04:00: Turned in 3 star job to Simon` appended to your log file

\> an hour later, you upload your packet log for someone else to sift through

Now looking at this log, you're only interested in seeing the packets for when a 3-star job is done for Simon. According to the log produced by the timestamp script, this fellow noted that was done at 13:51:12.432-04:00, so you scroll there and look around.

Without that log, you might try searching for "Simon", but you don't find it because `<npcname/>` or `<title name='NONE'/>` was used. So you go through occurrences of "NpcTalk", supposing you did not know you could have searched for "QuestClear".
Also, some of the quotes may be similar, but have different triggers. 1-star and 3-star completion rates might express equal disappointment in mistakenly placing their trust in you doing your job properly. Maybe this is one of those NPCs that have multiple quotes for one trigger.

_Aside: Wow it's a lot easier to go through Piaras.zip when you can just Ctrl+F._

**Comment by exectails** [comment]
OH~ so, more like a note feature, and not timestamps.

Though honestly, you simply shouldn't create 2 hours worth of logs in one file... no matter what you do, working with that file will be bothersome. The most efficient and clean way to log is walk up to the NPC, start logging, go through the dialog, stop logging, save it as "201606XX_XXXXXX - did something, ptj, responses, foobar, whatever helps you search for this.txt".

[My random log folder.](http://puu.sh/pmZGM/cdb508ff34.png)

**Comment by boilerplate5** [comment]
How many NPCs have actually been collected so far before I look into jumping on the PTJs? I don't want to end up doing ones that have already been captured but are as yet unimplemented.

**Comment by exectails** [comment]
You can see what has been logged here and on the Research forum: http://aura-project.org/forum/topic/173-ptj-dialogues/

**Comment by valdeza** [comment]
Assuming Walter's PTJ is approved for merging, that finishes up Dunbarton. I'm planning on eventually cleaning up/refactoring previous PTJs next, as promised; oh, the things we learned as Tir and Dun were implemented.

Now the problem is I have no research data on Bangor, so that's not something I can just code bit-by-bit like I have with Walter. With college classes going on, I highly doubt that I can dedicate time to sitting down and packet capturing Bangor PTJs. What I might be able to do is look at smaller bugs or issues that I can try resolving. Some of those could certainly take less time. Probably. (It's sort of hard to estimate how long it takes to code something...) Though then again, it's _just_ three NPCs. It's not like Tir or Dun.

Basically, at the time of writing, after doing that aforementioned cleanup PR I'm not sure whether to
- do Bangor PTJ research when I can or
- try resolving other issues and take down (or at least make significant progress) on Bangor PTJs during winter break.

Just sharing my thoughts after about a month of sudden silence. againsorry ;-;

**Comment by exectails** [comment]
At least the Bangor jobs are very simple if the Wiki is right. They seem to only have one kind of job per PTJ and it's all just collection or killing. Logging the messages should be enough to get all the information you need.

**Comment by valdeza** [comment]
Finished Bangor dialogue packet capture (or, at least I gathered what I could): [[link]](https://github.com/valdeza/aura/tree/resrch_ptj-dialogue/etc/ptj-dialogue/bangor)

Just in case I can't find time to work on this anytime soon.
(Suspiciously, this week doesn't look too busy. o.o )

**Comment by valdeza** [comment]
Bangor's written, just need to test Comgan and Sion... whenever this PC can be comfortably restarted.
After that, I'll write up Edern's PTJ for the future since I already have the dialogue for it. 

Looks like everything for G1 is just about at their final stages(?).
